player/lua/stats.lua

```diff
@@ -1776,19 +1776,25 @@ mp.register_event("video-reconfig",
 
 --  --script-opts=stats-bindlist=[-]{yes|<TERM-WIDTH>}
 if o.bindlist ~= "no" then
-    mp.command("no-osd set really-quiet yes")
-    if o.bindlist:sub(1, 1) == "-" then
-        o.bindlist = o.bindlist:sub(2)
-        o.no_ass_b0 = ""
-        o.no_ass_b1 = ""
-    end
-    local width = max(40, math.floor(tonumber(o.bindlist) or 79))
-    mp.add_timeout(0, function()  -- wait for all other scripts to finish init
+    mp.register_event("initialized", function()
+        mp.set_property("msg-level", "all=no,statusline=status")
+        mp.set_property("term-osd", "force")
+        mp.set_property_bool("msg-module", false)
+        mp.set_property_bool("msg-time", false)
+        if o.bindlist:sub(1, 1) == "-" then
+            o.bindlist = o.bindlist:sub(2)
+            o.no_ass_b0 = ""
+            o.no_ass_b1 = ""
+        end
+        local width = max(40, math.floor(tonumber(o.bindlist) or 79))
         o.ass_formatting = false
         o.no_ass_indent = " "
         o.term_size = { w = width , h = 0}
-        io.write(keybinding_info(false, true) .. "\n")
-        mp.command("quit")
+        mp.osd_message(keybinding_info(false, true), 42)
+        mp.add_timeout(0, function() -- flush status line on next tick
```

Comment:
Why is this timeout required? why not flush right away?

Please add a comment explaining why is this required.

Reply:
> Why is this timeout required? why not flush right away? 

osd message is updated by mpv core, not right away, we need to wait one tick for that.

> why is flush even required? shouldn't mpv flush everything on quit?

Mostly to flush status line without clearing it, which is standard flush on exit.

Reply:
Not sure I entirely follow, but these are very unobvious. Please add a comment explaining them.

Also, what about the "42"?

Reply:
> Also, what about the "42"?

timeout as any other, we are closing anyway, Would 1000 or 100 be better?

Reply:
If you don't need a specific timeout value, don't provide one. It's optional. Otherwise it's misleading and makes it harder to understand the intent.